pom.xml

```diff
@@ -1771,6 +1771,38 @@
             </modules>
         </profile>
 
+        <profile>
+            <id>jdk12</id>
+            <activation>
+                <jdk>12</jdk>
+            </activation>
+            <properties>
+                <surefire.jvm.args.java-version.lenient>
+                    -Djdk.attach.allowAttachSelf=true
+                    -Dcom.sun.xml.bind.v2.bytecode.ClassTailor.noOptimize=true
+                </surefire.jvm.args.java-version.lenient>
+                <!--
+                     When we use illegal-access=deny,
+                     we have to open the java.lang package to EasyMock,
+                     because cglib (used by EasyMock) requires to execute setAccessible(true)
+                     on ClassLoader.defineClass(...).
+                     See https://github.com/easymock/easymock/issues/235
+                     See https://hibernate.atlassian.net/browse/HSEARCH-3419
+                 -->
+                <surefire.jvm.args.java-version.strict>
+                    ${surefire.jvm.args.java-version.lenient}
+                    --illegal-access=deny
+                    --add-opens java.base/java.lang=ALL-UNNAMED
+                </surefire.jvm.args.java-version.strict>
+                <arquillian.wildfly.jvm.args.java-version>
+                    -Djdk.attach.allowAttachSelf=true
+                </arquillian.wildfly.jvm.args.java-version>
+            </properties>
+            <modules>
+                <!-- Nothing at the moment -->
+            </modules>
+        </profile>
+
```

Comment:
I think I would just have made the version of the above profile an open bound. Not sure it's worth having the exact same profile for N+ JDK.

Reply:
I didn't know we could do that. You're right, it would be better; will do.